deeppavlov/dataset_iterators/ner_few_shot_iterator.py

```diff
@@ -0,0 +1,162 @@
+                 filter_bi: bool = True,
+                 n_train_samples: int = 20,
+                 remove_not_targets = True,
+                 *args, **kwargs) -> None:
+        self.shuffle = shuffle
+        self.target_tag = target_tag
+        self.filter_bi = filter_bi
+        self.n_train_samples = n_train_samples
+        self.remove_not_targets = remove_not_targets
+
+
+        if self.target_tag is None:
+            raise RuntimeError('You must provide a target tag to NERFewShotIterator!')
+
+        self.random = Random(seed)
+
+        self.train = data.get('train', [])
+        self.valid = data.get('valid', [])
+        self.test = data.get('test', [])
+        self.split(*args, **kwargs)
+        self.data = {
+            'train': self.train,
+            'valid': self.valid,
+            'test': self.test,
+            'all': self.train + self.test + self.valid
+        }
+
+        self.n_samples = len(self.train)
+
+        if self.remove_not_targets:
+            self._remove_not_target_tags()
+
+        if self.filter_bi:
+            for key in self.data:
+                for n, (x, y) in enumerate(self.data[key]):
+                    self.data[key][n] = [x, [re.sub('(B-|I-)', '', tag) for tag in y]]
+
+        self.tag_map = np.zeros(self.n_samples, dtype=bool)
+        for n, (toks, tags) in enumerate(self.data['train']):
+            if self.filter_bi:
+                self.tag_map[n] = any(self.target_tag == tag for tag in tags if len(tag) > 2)
+            else:
+                self.tag_map[n] = any(self.target_tag == tag[2:] for tag in tags if len(tag) > 2)
+
+        self.marked_nums = None
+        self.unmarked_nums = None
+        self._sample_marked()
+
+    def _sample_marked(self):
+        np.zeros(len(self.data['train']), dtype=bool)
+        n_marked = 0
+        self.marked_mask = np.zeros(self.n_samples, dtype=bool)
+        while n_marked < self.n_train_samples:
+            is_picked = True
+            while is_picked:
+                n = np.random.randint(self.n_samples)
+                if not self.marked_mask[n]:
+                    is_picked = False
+                    self.marked_mask[n] = True
+                    if self.tag_map[n]:
+                        n_marked += 1
+
+        self.marked_nums = np.arange(self.n_samples)[self.marked_mask]
+        self.unmarked_nums = np.arange(self.n_samples)[~self.marked_mask]
+
+    def _remove_not_target_tags(self):
+        if self.remove_not_targets:
+            for key in self.data:
+                for n, (x, y) in enumerate(self.data[key]):
+                    tags = []
+                    for tag in y:
+                        if tag.endswith('-' + self.target_tag):
+                            tags.append(tag)
+                        else:
+                            tags.append('O')
+                    self.data[key][n] = [x, tags]
+
+    def get_instances(self, data_type: str = 'train') -> Tuple[tuple, tuple]:
```

Comment:
x, y = list(zip(*samples)) is of type Tuple[tuple, tuple]?

Reply:
Fixed